www/content/essays/no-build-step.md

```diff
@@ -0,0 +1,67 @@
+
+A more significant cost for using TypeScript is that the code running in the browser is not the code you wrote, which makes the browser's developer tools harder to use. When your TypeScript code throws an exception, you have to figure out how the stack trace (with its JavaScript line numbers, JavaScript function signatures, and so forth) maps to the TypeScript code that you wrote; when your JavaScript code throws an exception, you can click straight through to the source code, read the thing you wrote, and set a breakpoint in the debugger. This is *tremendous* DX. For many younger web developers who have never worked this way, it can be a revelatory experience.
+
+Build step advocates point out that TypeScript can generate [source maps](https://firefox-source-docs.mozilla.org/devtools-user/debugger/how_to/use_a_source_map/index.html), which tell your browser what TypeScript corresponds to what JavaScript, and that's true! But now you have *another* thing to keep track of—the TypeScript you wrote, the JavaScript it generated, and the source map that connects these two. The hot-reloading development server you're now dependent on will keep these up to date for you on localhost—but what about on your staging server? What about in production? Bugs that appear in these environments will be harder to track down, because you've lost a lot of information about where they come from. These are solvable problems, but they're problems you created; they are a cost.
+
+The htmx DX is very simple—your browser loads a single file, which in every environment is the exact same file you wrote. The tradeoffs required to maintain that experience are real, but they're tradeoffs that make sense for this project.
+
+## Enforced Clarity
+Modularization is one of the [honking great ideas](https://legacy.python.org/dev/peps/pep-0020/) of software. Modules make it possible to solve incredibly complex problems by breaking down code into well-contained substructures that solve smaller problems. Modules are really useful.
+
+Sometimes, however, you want to solve simple problems, or at least relatively simple problems. In those cases, it can be helpful *not* to use the building blocks of more complex software, lest you emulate their complexity without creating commensurate value. At its core, htmx solves a relatively simple problem: it adds a handful of attributes to HTML that make it easier to replace DOM elements using the declarative character of hypertext. Requiring that all the htmx code remain in a single file (again, around 3,500 lines of code) enforces a degree of intention on the library; there is a real pressure when working on the htmx source to justify the addition new  lines of code, a pressure which maintains a equilibrium of relative simplicity.
```

Comment:
> there is a real pressure when working on the htmx source to justify the addition new  lines of code, a pressure which maintains a equilibrium of relative simplicity.

1. "the addition [of] new lines of code" (missing word in square brackets)
2. "new lines of code" (remove extra space between "new" and "lines")
3. "a pressure which maintains a[n] equilibrium of relative simplicity" (missing character in square brackets)

Reply:
appreciate it, thank you!